includes/bp-messages/classes/class-bp-rest-sitewide-notices-endpoint.php

```diff
@@ -0,0 +1,1000 @@
+	 * @since 9.0.0
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_item' ),
+					'permission_callback' => array( $this, 'create_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::CREATABLE ),
+				),
+				'schema' => array( $this, 'get_item_schema' ),
+			)
+		);
+
+		// (?P<id>[\d]+) is the placeholder for the notice ID.
+		$notice_endpoint = '/' . $this->rest_base . '/(?P<id>[\d]+)';
+
+		register_rest_route(
+			$this->namespace,
+			$notice_endpoint,
+			array(
+				'args' => array(
+					'id' => array(
+						'description' => __( 'A unique numeric ID for the Sitewide notice.', 'buddypress' ),
+						'type'        => 'integer',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::READABLE ),
+				),
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'update_item' ),
+					'permission_callback' => array( $this, 'update_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+				),
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_item' ),
+					'permission_callback' => array( $this, 'delete_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::DELETABLE ),
+				),
+				'schema' => array( $this, 'get_item_schema' ),
+			)
+		);
+
+		$dismiss_endpoint = '/' . $this->rest_base . '/dismiss';
+
+		register_rest_route(
+			$this->namespace,
+			$dismiss_endpoint,
+			array(
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'dismiss_notice' ),
+					'permission_callback' => array( $this, 'dismiss_notice_permissions_check' ), // Anyone who can get items can dismiss them.
+				),
+				'schema' => array( $this, 'get_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Retrieve sitewide notices.
+	 *
+	 * @since 9.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response
+	 */
+	public function get_items( $request ) {
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
```

Comment:
In the `get_items_permissions_check`, you expect the context to be there, but here I see this conditional.

```suggestion
		$context = $request->get_param( 'context' );
```

Reply:
Good catch! Thanks a lot. I've updated the PR accordingly